### PR TITLE
chore: support multiple resources in controllerbuilder

### PR DESCRIPTION
**Pull request by justinsb**
Otherwise our types.generated.go was being overwritten

### CODE REVIEW

File: dev/tools/controllerbuilder/pkg/commands/generatetypes/generatetypescommand.go [hunk 1]
```diff
@@ -26,16 +26,49 @@ import (
...
+
+func (r *ResourceList) String() string {
+	var sb strings.Builder
+	for _, res := range *r {
+		fmt.Fprintf(&sb, "%s:%s", res.Kind, res.ProtoName)
+	}
+	return sb.String()
+}
+
+func (r *ResourceList) Set(s string) error {
```
Comment (jingyih) [inline dev/tools/controllerbuilder/pkg/commands/generatetypes/generatetypescommand.go:62]: This is super cool!

File: dev/tools/controllerbuilder/pkg/codegen/typegenerator.go [hunk 2]
```diff
@@ -41,45 +41,48 @@ var protoMessagesNotMappedToGoStruct = map[string]string{
...
+
+	g.visitedMessages = append(g.visitedMessages, messageDescriptor)
 
-	msgs, err := findDependenciesForMessage(msgDesc)
+	msgs, err := findDependenciesForMessage(messageDescriptor)
 	if err != nil {
 		return err
 	}
-	g.visitedMessages = append(msgs, msgDesc)
+	g.visitedMessages = append(g.visitedMessages, msgs...)
```
Comment (jingyih) [inline dev/tools/controllerbuilder/pkg/codegen/typegenerator.go:85]: Do we need to sort and remove duplicate after calling this function for multiple resources?

File: dev/tools/controllerbuilder/pkg/commands/generatetypes/generatetypescommand.go [hunk 3]
```diff
@@ -26,16 +26,49 @@ import (
...
+	tokens := strings.Split(s, ":")
+	if len(tokens) != 2 || tokens[0] == "" || tokens[1] == "" {
+		return fmt.Errorf("expected [KRMKind]:[ProtoResourceName], got %q", s)
+	}
+	*r = append(*r, Resource{
+		Kind:      tokens[0],
+		ProtoName: tokens[1],
```
Comment (jingyih) [inline dev/tools/controllerbuilder/pkg/commands/generatetypes/generatetypescommand.go:69]: nit: previously, users weren’t required to input a capitalized proto resource name, as the tool automatically called `capitalizeFirstRune` to handle this for them.

**Review by jingyih — COMMENTED** [review]
/lgtm

cc @acpana

File: dev/tools/controllerbuilder/pkg/commands/generatetypes/generatetypescommand.go (hunk 1 above)
Comment (yuwenma) [reply dev/tools/controllerbuilder/pkg/commands/generatetypes/generatetypescommand.go:62]: +100!

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve
/hold

Do you mind updating the generate.sh and 2-define-apis.md in the same PR? Several external contributors are following the guide, merging this PR may confuse them and introduce some unnecessary overhead.

File: dev/tools/controllerbuilder/pkg/codegen/typegenerator.go (hunk 2 above)
Comment (justinsb) [reply dev/tools/controllerbuilder/pkg/codegen/typegenerator.go:85]: Good call.  This list is only used when we write the messages, so I just sort and dedup there now.

File: dev/tools/controllerbuilder/pkg/commands/generatetypes/generatetypescommand.go (hunk 3 above)
Comment (justinsb) [reply dev/tools/controllerbuilder/pkg/commands/generatetypes/generatetypescommand.go:69]: Yeah, I think that was a bit too user-friendly :-)

**Comment by justinsb** [comment]
Fixed problems, PTAL @yuwenma

**Pull request merged**